### ISSUE TITLE
Provide additional sound samples which are not spoken word

### ISSUE THREAD

**Issue by capital-G** [supercollider/supercollider#5604]
## Motivation

I think it would be good to have 2-3 other sound samples than the infamous `a11wlk01.wav` or not so infamous `a11wlk01-44_1.aiff` which are lofi spoken word recordings.
I currently working on a machine listening library and having only spoken word samples is not a good way to show how the library could be used w/o providing sound examples or generating ones via SynthDef.

## Description of Proposed Feature

I understand that choosing a sample for such a sonically non-determined and open platform as SC is difficult and could spawn discussions, leaving legal/license stuff aside. On the other hand: we also have 2 spoken word examples (one(?) of it in english) shipped with every SC installation.

## Plan for Implementation

We could simply define some boundaries (license, recording-quality, length) and choose a random example from https://freesound.org/ which could be included from thereon on every installation.
Maybe having one percussive, one high-quality field-recording and one tonal example would fit some use-cases and definitely extend the existing examples.

I could take care of this and it will not take too long, but this depends on the discussion attached to this.

**Comment by fmiramar** [comment]
Definitely a great improvement to do. I think that if this change could also be associated with some documentation improvements it would be really interesting. 

>  Maybe having one percussive, one high-quality field-recording and one tonal example would fit some use-cases and definitely extend the existing examples.

My suggestions on some other sounds and associated documentations changes that would be nice to have:

- drum loop - Onsets / BeatTrack / Other Analysis UGens; Patterns examples
- ~3 short and similar shakers samples + ~3 other short and similar percussive sounds (kick, hihat, etc) -- `Buffer` or `PathName` doc -- Eli Fieldsteel example about recursive reading samples on folders; Patterns examples
- one orchestral chord -- various PV UGens examples 
- one male + one female singing voices (talking voice already exists) -- Formant extraction methods (Cepstrum, PV_CopyPhase, etc), PitchShift examples
- one polyphonic pop song excerpt for MIR examples -- Analysis UGens (BeatTrack sound example seems to be missing)
- Unclear pitch/noise/harmonic/inharmonic example (tamtam, gong, etc) - BufRd waveshaping examples, SMS examples 

I would careful about the size of files, especially the field recording which tends to be large (some popular quarks already provide those like the ATK).

**Comment by capital-G** [comment]
Would this be something which could be included in 3.13 https://github.com/supercollider/supercollider/milestone/24 @telephon ?

At least it would be nice to have an API implemented which allows to access existing samples in a documented and autocompleted manner and allows to add new samples in the future (e.g. following releases of 3.13).

I always need to search for `Buffer.read(s, Platform.resourceDir +/+ "sounds/a11wlk01-44_1.aiff")` in the `Buffer` documentation and something like `ExampleBuffers.nasa` would be a nice abstraction layer for beginners and for convenience (and would allow for documentation on what these samples actually are).

But should it return a Path to a Buffer or a (new?) Buffer? `ExampleBuffers` sound like it would return Buffer, but a Paths allow more general handling, yet I need information such as number of channels which I don't know w/o looking at the file or documentation. Maybe providing an `ExampleBuffers` and an `ExampleBufferPaths`?

**Comment by telephon** [comment]
Good idea. I wouldn't call the class `ExampleBuffer`, because it would limit its use to "examples". Users should be able to register buffers easily.

**Comment by jamshark70** [comment]
> At least it would be nice to have an API implemented which allows to access existing samples in a documented and autocompleted manner and allows to add new samples in the future (e.g. following releases of 3.13).

> Users should be able to register buffers easily.

Let's be careful here -- there is not much time before the milestone.

A new class might be feasible to add within the available time, if it's limited to providing handles to known example sound files. But that's at risk of being bloat-y... a whole class just to return paths, to example sound files that people won't be using in production anyway... I dunno, I'm not certain that would be worth the weight. IMO editor snippets would be better, just insert the code string upon a key binding.

Which I think is why telephon suggested a more general audio-file library class. But, to do that right, we should gather user requirements and go through a proper design cycle. For example, would users want to be able to tag entries? Where would the tags come from? (Would we try to decode metadata?) What is the relationship between library entries and directory structure? (That is, if you access an entry in the library by, say, `AudioFiles.at(\path, \to, \some, \directory, \and, "thefile.wav")`, how is this a significant improvement over "/path/to/some/directory/and/thefile.wav"? There may be improvement indeed! But I don't think we can assume that "shiny new toy in the class library" = significant workflow improvement.)

I'm extremely skeptical of the value of rushing this sort of design process. We have gotten into trouble in the past by saying "Good idea" and not thinking it through. IMO this is a complex enough requirement that it probably should go through an RFC process.

> But should it return a Path to a Buffer or a (new?) Buffer?

Absolutely not a Buffer. Let the user decide what to do with the path (but as noted, there are other ways to provide paths, e.g. snippets, especially if it's a small number of them).

**Comment by redFrik** [comment]
i agree. let's not rush this. a few more sound files would be good, but there are many things to consider before adding a new convenience like this.
e.g. should it support other types of media (reference: `Image.open(SCDoc.helpSourceDir +/+ "images/Swamp.png");`) and then need a more generic name (Media, Resources, Data...)?

**Comment by telephon** [comment]
> I'm extremely skeptical of the value of rushing this sort of design process.

Yes, definitely will take time.


> e.g. should it support other types of media

Yes, it could be just a matter of paths actually.

I think that the issue is triggered by:
- the lack of an easy access to resources in terms of paths in general (what is there and where is it)
- the integration of data into code that is valid for more than one person (how do I write meaningful helpfiles, and share code)

**Comment by telephon** [comment]
... but _this_ issue here might indeed better be solved by just adding one non spoken word sample to the default sounds. This is an act of high artistic responsibility (only half tongue in cheek)! But I'd say make suggestions (perhaps music that is not attributable to a single culture) …

**Comment by capital-G** [comment]
I am not sure if you allow modifications/extensions to the standard library within one major release, that was my motivation to maybe *rush* in a simple class as

```supercollider
BufferExamplePaths {
  *nasa { ^Platform.resourceDir +/+ "sounds/a11wlk01-44_1.aiff" }

  *baby { ^Platform.resourceDir +/+ "sounds/a11wlk01.wav" }
}
```

so the often used (and probably most of the time copied from someplace else)

```supercollider
b = Buffer.read(s, Platform.resourceDir +/+ "sounds/a11wlk01-44_1.aiff");
```

becomes a much more remember-able and autocomplete-able

```supercollider
b = Buffer.read(s, BufferExamplePaths.nasa);
```  

An additional benefit: The samples could be attributed and described via documentation and we could add additional, non-spoken-word samples in future releases in a standardized and documented manner.

> Good idea. I wouldn't call the class `ExampleBuffer`, because it would limit its intended use to "examples". Users should be able to register buffers easily.

I really enjoy this idea but this could also be something for a Quark - like a Sample database?
This is more complicated as it introduces a state which needs to be stored in a somehow secure manner on the local storage - using a simple one-line-per-sample file is rather limited in information to be stored, problematic on dynamic generation of fields, needs to support [different kind of line breaks for different OS](https://stackoverflow.com/a/39259747/3475778) and then we have not talked about race conditions.
As the I/O capabilities of SC are rather limited (e.g. you can not specify the encoding to use while reading/writing to a file and SC has still no official complete JSON support - although I written a basic SC event to JSON serializer [here](https://github.com/capital-G/quarks-web/blob/12233b4e47cac8574b85433071a05d8775483d59/quarkToJson.scd#L5-L32)) I think this needs some more engineering and is indeed a bit of a bigger task with some design decisions to be considered.
Maybe it would also be nice to offer a set of Samples on the internet, like Sample-Quarks. But still, I think it is better that this is developed as a Quark with different iterations and experiments and once stable it can be included to the standard library.

> a whole class just to return paths, to example sound files that people won't be using in production anyway

Well, in the end most of programming is just convenience and syntax sugar but I think the convenience things are also important - and while they are probably not used in production often that does not mean that they are not used. They are often used in the documentation and also often used in teaching situations as these are the only 2 files which everyone has access to when installing SC (Its a pity though that both of them have a lofi-quality).
Of course there are things such as DDWSnippets but I think if you know DDW already you are an advanced user and could write the class defined above yourself but it still would not allow for an easy integration in a teaching situation.

> e.g. should it support other types of media (reference: `Image.open(SCDoc.helpSourceDir +/+ "images/Swamp.png");`) and then need a more generic name (Media, Resources, Data...)?

I was not aware that SC shipped with any more media data than those 2 samples - is it only this picture or more? But for me it would be sufficient to not search for `a11wlk01-44_1.aiff` anymore, so for me Buffer-only would be fine.

> But I'd say make suggestions (perhaps music that is not attributable to a single culture) …

Maybe something like a violin? It allows for discrete and continuous pitch and different playing styles while also having rich overtones and while the violin originates in western culture it is also used in persian and indian music.
A trombone would also be an alternative.

I find this *representative* kind of thinking rather difficult I must admit as I am no expert outside of my tradition and while an openness and variety is important we also have already the english language as a default language within SC which defines a cultural context and I think it is more important to state this bias (and be open to contributions to introduce multiple biases) than trying to hide it by trying to represent something *unifying*.

**Comment by capital-G** [comment]
> I'd also be cautious about introducing a new class to point to examples

Could you elaborate your concerns?
To me it has the advantages of providing auto completion and documentation. It also allows to add logic beyond symlinks, such as deprecations, symlinks or caching from the internet.

As I work primarily in Python I can give some examples of libraries which provide access to such files

* **librosa**: A python audio library, see https://librosa.org/doc/main/recordings.html and https://librosa.org/doc/main/generated/librosa.util.example.html# which can be listened https://github.com/librosa/data/tree/main/audio

  ```python
  y, sr = librosa.load(librosa.example('brahms'))
  ```

* **scikit-image**: https://scikit-image.org/docs/dev/api/skimage.data.html#skimage.data.astronaut

  ```python
  skimage.data.astronaut()
  ```

* **torchaudio** https://pytorch.org/audio/main/datasets.html#librimix

  ```python
  torchaudio.datasets.LibriMix()
  ```
* **sklearn**: https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_california_housing.html#sklearn.datasets.fetch_california_housing

  ```python
  sklearn.datasets.fetch_california_housing()
  ```

As SuperCollider lacks the concept of namespaces a class would be the closest equivalent.

**Comment by dyfer** [comment]
> > I'd also be cautious about introducing a new class to point to examples
> 
> Could you elaborate your concerns? To me it has the advantages of providing auto completion and documentation. It also allows to add logic beyond symlinks, such as deprecations, symlinks or caching from the internet.

Yes, I recognize that it has advantages like autocompletion (not sure if caching from the internet applies here though). 

IMO this functionality would belong better in path tools (probably `PathName`) than in the `Buffer`. E.g. having this as path tool would allow you to use it in a `SoundFile`. 
Also, it feels cleaner to me to replace path, as opposed to an object, in order to swap examples for one's own soundfiles, but that's my personal opinion. 

In any case... in _this_ issue we should really be discussing adding non spoken word examples. If we want to discuss exposing examples as objects, let's open a separate issue for that.

**Comment by capital-G** [comment]
I created #5859 for discussion about the class, allowing for discussing the Examples here.

The examples could be saved in FLAC format, allowing for a reduced footprint.

To throw some examples into the discussion

Link/Name | Description | Comment
--- | --- | ---
[Bird Whistling, Robin, Single, 13.wav](https://freesound.org/people/InspectorJ/sounds/456440/#) | Adds non-human example, rather high pitched | From librosa examples
[solo trumpet -06in F - 90bpm.wav](https://freesound.org/people/sorohanro/sounds/77711/#) | Lots of overtones (nice for filtering) and rather dry - uses traditional scales | From librosa examples
[Santur Phrase II](https://freesound.org/people/nsmusic/sounds/258079/#) | `Pluck` like, lots of transients but nice decay. Good for transient detection and reverb experiments | 
[fiddle05.flac](https://freesound.org/people/dobroide/sounds/11469/#) | Non-discrete pitches

Please add some, here is the markdown template

```markdown
Link/Name | Description | Comment
--- | --- | ---
a | b | c
```

 Maybe collect a couple of examples and create a voting system for them and add them accordingly?

**Comment by prko** [comment]
Some Korean instrument samples in stereo formats:
http://en.catsnu.com/Project/GugakVSTi.aspx

Clicking the `EXS24` or `Live Sampler` link of any instrument will open a Dropbox webpage to download that instrument. 
I love pyeongyeong, yangeum and janggu.

Link/Name | Description | Comment
--- | --- | ---
[Pyeongyeong](https://www.dropbox.com/sh/camtmcoby56roxh/AAB6INfWnX_75XHepC14moXia?dl=0) | Pure, bell‑like, dignified; quick decay with a slightly dry edge
[Yangeum](https://www.dropbox.com/sh/r7tpzqoi1q6krd4/AADRW7VXJBvojT4IHjykdx-Ba?dl=0) | Bright, metallic, percussive; crisp attack with shimmering overtones
[Janggu](https://www.dropbox.com/sh/nlckvgrxjzp8ib7/AABeF4hBc5h5U3lR_x25AqQZa?dl=0) | Dual timbre: deep, resonant booms (low head) + sharp, cracking slaps (high head)

**Comment by HotwheelsSisyphus** [comment]
It just occurred to me that it might be a good idea to have a more clean/hi-fi spoken word sample as well. Currently we have an infant and a walkie-talkie distorted/filtered adult...

**Comment by HotwheelsSisyphus** [comment]
Not sure if this is OT (because it is _not_ not spoken word), but following up on my last remark ("it might be a good idea to have a more clean/hi-fi spoken word sample as well"), I looked over some of the public domain audiobooks on archive.org to find candidates from which to extract a shorter "clean" spoken word example. These are generally a mixed bag, not all of them sound terribly professional (room resonances and such). They also tend to be mp3. But here's one that seems ok to my ears, fairly dry and articulate. Both text and audio are in the public domain, from what I understand.

https://archive.org/details/adrift_in_the_unknown_2311_librivox/adriftintheunknown_01_cook_128kb.mp3

Maybe someone with intact hearing and/or more experience working with recorded speech could give their two cents?

**Comment by JordanHendersonMusic** [comment]
Does any one think they might like to do this?  I wouldn't bother with voting personally, as long as they are small, something is better than the nothing we have right now.

**Comment by HotwheelsSisyphus** [comment]
> Does any one think they might like to do this? 

Imho it's currently not very clear what "this" is, exactly...

**Comment by JordanHendersonMusic** [comment]
IMO, 'this' is about 5 new short sound files with open licenses exhibiting a wide range of sonic characteristics to be used as examples and test cases by users and documentation writers.
